scripts/analyze

```diff
@@ -0,0 +1,168 @@
+        data = json.load(f)
+    if 'depends' in data:
+        depends = data['depends']
+        dependencies = add_to_list(dependencies, depends)
+        for i in depends:
+            l = find_deps(i)
+            dependencies = add_to_list(l, dependencies)
+    if 'targets' in data:
+        found = False
+        for i in data['targets']:
+            if i == board:
+                found = True
+                break
+        if not found:
+            return
+    if 'zephyr_conf' in data:
+        z_conf = data['zephyr_conf']
+        if 'all' in z_conf:
+            zephyr_config = add_to_list(zephyr_config, z_conf['all'])
+        if board in z_conf:
+            zephyr_config = add_to_list(zephyr_config, z_conf[board])
+    if 'zjs_config' in data:
+        zjs_config = add_to_list(zjs_config, data['zjs_config'])
+    if 'jrs_config' in data:
+        jrs_config = add_to_list(jrs_config, data['jrs_config'])
+    if 'src' in data:
+        src = data['src']
+        if isinstance(src, list):
+            src_list = add_to_list(src_list, src)
+        else:
+            if 'common' in src:
+                src_list = add_to_list(src_list,src['common'])
+            for key, value in src.items():
+                if key != "common":
+                    # var x = require(module).key
+                    off_req = re.search(r"var .*require\([\'\"]%s[\"\']\)\.%s" % (data["require"], key), text)
+                    if off_req:
+                        if 'src' in value:
+                            src_list = add_to_list(src_list, value['src'])
+                        if 'zjs_config' in value:
+                            zjs_config = add_to_list(zjs_config, value['zjs_config'])
+                    else:
+                        # var x = require(module)
+                        # var y = x.y
+                        match_name = re.search(r"var .*require\([\'\"]%s[\"\']\)" % data["require"], text)
+                        var_name = match_name.group(0).split(' ')[1]
+                        match_use = re.search(r"%s\.%s" % (var_name, key), text)
+                        if match_use:
+                            if 'src' in value:
+                                src_list = add_to_list(src_list, value['src'])
+                            if 'zjs_config' in value:
+                                zjs_config = add_to_list(zjs_config, value['zjs_config'])
+
+with open(script) as js:
+    text = js.read()
+
+for file in os.listdir("src/"):
+    if file.endswith(".json"):
+        with open("src/" + file) as f:
+            data = json.load(f)
+        if 'require' in data:
+            if re.search(r"require\([\'\"]%s[\"\']\)" % data["require"], text):
+                find_deps(data["require"])
+        if 'constructor' in data:
+            if re.search(r"%s\(.*" % data["constructor"], text):
+                find_deps(data["module"])
+
+find_deps(board)
+if force != ['']:
+    for i in force:
+        find_deps(i)
```

Comment:
Since lists are mutable, you could just pass dependencies into find_deps and not have to make it a global.